main.cpp

```diff
@@ -131,287 +172,412 @@ struct CarWash
-//     5) Tournaments won (int)
         int tournamentsWon = 0;
-//     3 things it can do:
-//     1) drive
-        float drive(float power, float accuracy); // returns distance to tee
-//     2) putt
-        float putt(float power, float accuracy); // returns distance to tee
-//     3) chip
-        float chip(float power, float accuracy); // returns distance to tee
+
+        float drive(float power, float accuracy, float distanceToTee);
+        float putt(float power, float accuracy, float distanceToTee);
+        float chip(float power, float accuracy, float distanceToTee);
     };
-// 5 properties:
-//     1) number of holes (int)
-    int numHoles = 18;
-//     2) number of golfers per week (double)
-    double golfersPerWeek = 15000;
-//     3) length, in yards (float)
-    float courseLength = 6500;
-//     4) number of golfers per party (int)
-    int golfersPerParty = 4;
-//     5) members-only (or not) (bool)
-    bool membersOnly = false;
-// 3 things it can do:
-//     1) host a tournament
+
     void hostTournament(int numParticipants, int tournamentDate, float purse, float firstPlacePercentage,         float secondPlacePercentage, float thirdPlacePercentage); 
-//     2) suspend play for weather reasons
-    bool suspendPlay(float duration); // returns false to suspend play for the specific duration; would then return back to true, resuming play 
-//     3) change hole location on green
+    bool suspendPlay(float duration);
     void changeHoleLocation(int holeNum, float latitude, float longitude, Golfer tester); 
 
     Golfer golfPro;
 };
 
-// Thing 4) library
+float GolfCourse::Golfer::drive(float power, float accuracy, float distanceToTee)
+{
+    return distanceToTee - (power * distanceToTee * .75f * accuracy);
+}
+
+float GolfCourse::Golfer::putt(float power, float accuracy, float distanceToTee)
+{
+    return distanceToTee - (power * distanceToTee * .25f * accuracy);
+}
+
+float GolfCourse::Golfer::chip(float power, float accuracy, float distanceToTee)
+{
+    return distanceToTee - (power * distanceToTee * .50f * accuracy);
+}
+
+void GolfCourse::hostTournament(int numParticipants, int tournamentDate, float purse, float firstPlacePercentage, float secondPlacePercentage, float thirdPlacePercentage)
+{
+    std::cout<<"It's a golf tournament!"<<std::endl;
+    std::cout<<numParticipants<<" will gather on "<<tournamentDate<<"to compete for "<<purse<<" in winnings!"<<std::endl;
+    std::cout<<"1st prize will win: "<<purse * firstPlacePercentage<<std::endl;
+    std::cout<<"2nd prize will win: "<<purse * secondPlacePercentage<<std::endl;
+    std::cout<<"3rd prize will win: "<<purse * thirdPlacePercentage<<std::endl;
+}
+
+bool GolfCourse::suspendPlay(float duration)
+{
+    if(duration == 0.0f) 
+    {
+        std::cout<<"No suspension necessary!";
+        return 1;
+    }
+    else
+    {
+        std::cout<<"Play will be suspended for "<<duration<<std::endl;
+        return 0;
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise